vm/devices/net/mana_driver/src/mana.rs

```diff
@@ -329,11 +330,12 @@ impl<T: DeviceBacking> Vport<T> {
         cpu: u32,
     ) -> anyhow::Result<BnicEq> {
         let mut gdma = self.inner.gdma.lock().await;
-        let mem = gdma
-            .device()
-            .host_allocator()
+
+        let dma_client = gdma.device().get_dma_client().context("Failed to get DMA client from device")?;
```

Comment:
We talked about this before, but this `get_dma_client` should not return an Option, IE the method should not be faillable. Every device will want a dma client.